### ISSUE TITLE
Impossible to import QuTiP in Google's colab (used to work a couple of weeks ago)

### ISSUE THREAD

**Issue by ziofil** [qutip/qutip#1008]
I'm teaching a course where we use QuTiP, but recently we cannot import QuTiP anymore : /

The procedure to reproduce the error is to open a new notebook in [colab](http://colab.research.google.com/), install QuTiP with 

    !pip install qutip

and then just

    import qutip

The error in this case is 

    AttributeError                            Traceback (most recent call last)
    /usr/local/lib/python3.6/dist-packages/qutip/__init__.py in <module>()
         46     __IPYTHON__
    ---> 47     qutip.settings.ipython = True
         48 except:

    AttributeError: module 'qutip' has no attribute 'settings'

    During handling of the above exception, another exception occurred:

    AttributeError                            Traceback (most recent call last)
    <ipython-input-6-e01d5608f34f> in <module>()
    ----> 1 import qutip

    /usr/local/lib/python3.6/dist-packages/qutip/__init__.py in <module>()
         47     qutip.settings.ipython = True
         48 except:
    ---> 49     qutip.settings.ipython = False
         50 
         51 # -----------------------------------------------------------------------------

    AttributeError: module 'qutip' has no attribute 'settings'

**Comment by nathanshammah** [comment]
I tried it on Colab and obtained this error if importing a single function,
which seems related to the init, 

```
AttributeError                            Traceback (most recent call last)
/usr/local/lib/python3.6/dist-packages/qutip/__init__.py in <module>()
     46     __IPYTHON__
---> 47     qutip.settings.ipython = True
     48 except:

AttributeError: module 'qutip' has no attribute 'settings'

During handling of the above exception, another exception occurred:

AttributeError                            Traceback (most recent call last)
1 frames
/usr/local/lib/python3.6/dist-packages/qutip/__init__.py in <module>()
     47     qutip.settings.ipython = True
     48 except:
---> 49     qutip.settings.ipython = False
     50 
     51 # -----------------------------------------------------------------------------
```
AttributeError: module 'qutip' has no attribute 'settings'

The strange thing is that that bit of the init has not been touched for years,  https://github.com/qutip/qutip/blame/master/qutip/__init__.py#L45

I don't have an immediate fix (@quantshah might know more about colab) but a workaround could be using [My Binder](https://mybinder.org/), if that is a viable option for you. 

To set it up, you need to copy a `requirements.yml` file in your repository and activate My Binder by copying the link into their homepage.
Right now, in the master of qutip-notebooks we get an issue with scipy's `IndexMixin`  
https://github.com/theislab/anndata/issues/146

But, from my [notebooks](https://github.com/nathanshammah/notebooks) repository it seems to work. Copying the environment file and copy-pasting the My Binder link should work. You can copy-paste a given branch in mybinder.org, such that it does not keep rebuilding the environment completely from scratch, something that will happen if it finds a new commit in that branch. Hope this helps. 

.

**Comment by nathanshammah** [comment]
Copy also the `apt.txt` as from my notebooks folder if you want latex in plots.

**Comment by ziofil** [comment]
Thank you, I will look into these alternatives.

**Comment by quantshah** [comment]
Hey,

I faced similar issues and it's due to Scipy. I am meaning to open an issue
about this but try

!pip install Scipy==1
!pip install qutip

Worked for me a three days back in June 1.

**Comment by nathanshammah** [comment]
@ziofil, update your colab notebook to have the line
```
!pip install scipy==1.2.1
!pip install qutip
```
and it will work, as @quantshah mentioned. 
QuTiP will find already the SciPy 1.2.1 requirement and will avoid keeping 1.3.
It is an issue with `IndexMixin` relocation in SciPy 1.3 (released this May 17), which has been changed from `scipy.sparse.sputils` to `scipy.sparse._index`.
I tested this also in My Binder and this works too, by setting in the `environment.yml` file, 
[in line 9,](https://github.com/nathanshammah/interactive-notebooks/blob/master/environment.yml#L9)
from
```
- scipy
```
to 
```
- scipy==1.2.1
```
I will open a PR with this patch for `qutip-notebooks` for the time being. Once the next official version of QuTiP is released, this problem will not arise, as this issue has been taken care of by @Ericgig in PR #999 by updating scipy `IndexMixin`, now under private use, with scipy `get_index_dtype`.

**Comment by nathanshammah** [comment]
Solved by QuTiP 4.4.0 PyPI release. It proved quite slow to build the wheels on my try on Colab (5 minutes), we can open another issue in case this is standard.
